modules/sr/robot3/arduino.py

```diff
@@ -0,0 +1,289 @@
+            # splat in both the digital and analog pins
+            *(Pin(index, supports_analog=False) for index in range(2, 14)),
+            *(Pin(index, supports_analog=True) for index in AnalogPins),
+        )
+
+        # Connect the devices to the pins.
+        # NOTE: the names here correspond to the names given to devices in Webots
+        # and, in some places, the keyboard controller.
+        self._pins[2]._connect_device(Microswitch(webot, 'back bump sensor'))
+        self._pins[3]._connect_device(Led(webot, 'led 1', pin_num=3))
+        self._pins[4]._connect_device(Led(webot, 'led 2', pin_num=4))
+        self._pins[AnalogPins.A0]._connect_device(DistanceSensor(webot, 'Front Left DS'))
+        self._pins[AnalogPins.A1]._connect_device(DistanceSensor(webot, 'Front Right DS'))
+        self._pins[AnalogPins.A2]._connect_device(DistanceSensor(webot, 'Left DS'))
+        self._pins[AnalogPins.A3]._connect_device(DistanceSensor(webot, 'Right DS'))
+        self._pins[AnalogPins.A4]._connect_device(DistanceSensor(webot, 'Front DS'))
+        self._pins[AnalogPins.A5]._connect_device(DistanceSensor(webot, 'Back DS'))
+        # NOTE: these pins don't exist on a real arduino.
+        self._pins[AnalogPins.A6]._connect_device(
+            PressureSensor(webot, 'finger pressure left'),
+        )
+        self._pins[AnalogPins.A7]._connect_device(
+            PressureSensor(webot, 'finger pressure right'),
+        )
+
+    def identify(self) -> BoardIdentity:
+        """
+        Get the identity of the board.
+
+        The asset tag of the board is the serial number.
+
+        :return: The identity of the board.
+        """
+
+        return BoardIdentity(
+            manufacturer="Student Robotics",
+            board_type="SRduino",
+            asset_tag=self._serial_num,
+            sw_version='1.0',
+        )
+
+    @property
+    def pins(self) -> tuple[Pin, ...]:
+        """
+        The pins on the Arduino.
+
+        :return: A tuple of the pins on the Arduino.
+        """
+        return self._pins
+
+    def command(self, command: str) -> str:
+        """
+        Send a command to the board.
+        NOTE This is currently not implemented in the simulator.
+
+        :param command: The command to send to the board.
+        :return: The response from the board.
+        """
+        return ''
+
+    def map_pin_number(self, pin_number: int) -> str:
+        """
+        Map the pin number to the the serial format.
+        Pin numbers are sent as printable ASCII characters, with 0 being 'a'.
+
+        :param pin_number: The pin number to encode.
+        :return: The pin number in the serial format.
+        :raises ValueError: If the pin number is invalid.
+        """
+        try:  # bounds check
+            self.pins[pin_number]._check_if_disabled()
+        except (IndexError, IOError):
+            raise ValueError("Invalid pin provided") from None
+        return chr(pin_number + ord('a'))
```

Comment:
These appear to be unused (and aren't part of the documented API)
```suggestion
```

Reply:
As above, this function is present in the kit API. `command` is part of the extended SR firmware.